packages/bench-trail/runner.js

```diff
@@ -0,0 +1,333 @@
+  })
+}
+
+function reportFasterOpsperSec (suites) {
+  const sorted = suites.sort((a, b) => b.median - a.median)
+  const first = sorted[0]
+  const second = sorted[1]
+
+  const diffMedian = (first.median - second.median) / first.median * 100
+
+  console.log(
+    ` Speed: %s was faster by %s (%s vs %s)`,
+    chalk.yellow(first.name),
+    chalk.white.bold(diffMedian.toFixed(2) + '%'),
+    chalk.yellow(fnumber(first.median) + 'Hz'),
+    chalk.yellow(fnumber(second.median) + 'Hz')
+  )
+}
+
+function runBenchmark (suiteBenchmark) {
+  return new Promise((resolve, reject) => {
+    const suite = new Benchmark.Suite()
+
+    const asyncLabel = suiteBenchmark.async ? 'ASYNC' : 'SYNC'
+    console.log(
+      '\n%s %s [%s]\n',
+      chalk.white.bold('Benchmarking:'),
+      chalk.bold(suiteBenchmark.name),
+      asyncLabel
+    )
+
+    const isASync = !!suiteBenchmark.async
+    const benchmarkMethod =
+      isASync === true
+        ? lib.benchmark.async(suiteBenchmark.benchmark)
+        : suiteBenchmark.benchmark
+
+    for (let index = 0; index < iterations; index++) {
+      suite.add(`${index + 1} ${suiteBenchmark.name}`, {
+        defer: isASync,
+        fn: benchmarkMethod
+      })
+    }
+
+    // add listeners
+    suite
+      .on('cycle', function (event) {
+        console.log('', String(event.target))
+      })
+      .on('error', reject)
+      .on('complete', function () {
+        const benchmarks = Array.from(this)
+        const hzSet = benchmarks
+          .map(benchmark => benchmark.hz)
+          .sort((a, b) => a - b)
+        const hzSum = sum(hzSet)
+
+        const average = hzSum / hzSet.length
+        const median = middle(hzSet)
+
+        console.log(
+          '\n Ran %s (%s times) with an average of %s ops/sec',
+          chalk.yellow(suiteBenchmark.name),
+          chalk.yellow(iterations),
+          chalk.yellow(fnumber(average))
+        )
+        console.log('  Fastest: %s ops/sec', fnumber(Math.max(...hzSet)))
+        console.log('  Average: %s ops/sec', chalk.bold(fnumber(average)))
+        console.log('  Median : %s ops/sec', chalk.white.bold(fnumber(median)))
+        console.log('  Slowest: %s ops/sec', fnumber(Math.min(...hzSet)))
+
+        resolve(
+          Object.assign({}, suiteBenchmark, {
+            average,
+            median
+          })
+        )
+      })
+      // run async
+      .run({ async: true })
+  })
+}
+
+function fnumber (x) {
+  return Math.floor(x)
+    .toString()
+    .replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+}
+
+function sum (values) {
+  return values.reduce((acc, val) => acc + val)
+}
+
+function middle (values) {
+  var len = values.length
+  var half = Math.floor(len / 2)
+
+  if (len % 2) {
+    return (values[half - 1] + values[half]) / 2.0
+  } else {
+    return values[half]
+  }
+}
+
+function bytesToKb (bytes) {
+  return Math.round(bytes / 1024 * 100) / 100
+}
+
+function runGC (val) {
```

Comment:
all memory related code is not being used ATM, I left it here because it took me a while to figure some things out. will look back into it with more time to reintegrate it